core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -521,8 +521,18 @@ object SparkSubmit {
         sysProps.put("spark.yarn.isPython", "true")
       }
       if (args.principal != null) {
-        require(args.keytab != null, "Keytab must be specified when the keytab is specified")
-        UserGroupInformation.loginUserFromKeytab(args.principal, args.keytab)
+        require(args.keytab != null, "Keytab must be specified when principal is specified")
+        if (!new File(args.keytab).exists()) {
+          throw new SparkException(s"Keytab file: ${args.keytab} does not exist")
+        } else {
+          // Add keytab and principal configurations in sysProps to make them available
+          // for later use (e.g. by spark sql). These Configurations will be set as
```

Comment:
Let's be more specific on the use case of spark sql. Let's say the isolated class loader used to talk to HiveMetastore will use these settings.